https://github.com/haiwen/seafile-client/pull/1452
https://github.com/haiwen/seafile-client/pull/1314

From b5b1719c5b69c5c9f2396782ced71523d013446d Mon Sep 17 00:00:00 2001
From: orbea <orbea@riseup.net>
Date: Sat, 8 Apr 2023 18:25:37 -0700
Subject: [PATCH] rsa: Fix the build with libressl 3.7

Newer versions of libressl no longer need this legacy code path.

Patch from OpenBSD:

https://github.com/openbsd/ports/blob/26afb45924c9faa3bbd31e76a21130bbad474994/net/seafile/client/patches/patch-src_utils_rsa_cpp
---
 src/utils/rsa.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/utils/rsa.cpp b/src/utils/rsa.cpp
index 350b79012..943a55f3b 100644
--- a/src/utils/rsa.cpp
+++ b/src/utils/rsa.cpp
@@ -13,7 +13,8 @@ namespace {
 
 /* Forward compatibility functions if libssl < 1.1.0. */
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || \
+   (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x2070000fL)
 
 int RSA_set0_key(RSA *r, BIGNUM *n, BIGNUM *e, BIGNUM *d)
 {
